src/library/basesqltablemodel.cpp

```diff
@@ -419,6 +442,22 @@ void BaseSqlTableModel::setSort(int column, Qt::SortOrder order) {
             m_sortColumns.removeLast();
         }
     }
+    
+    // Write new sortColumns order to user settings
+    QString val;
+    QTextStream out(&val);
+    for (SortColumn& sc : m_sortColumns) {
+        int ccColumn = sc.m_column - m_tableColumns.size() + 1;
+        out << m_trackSource->columnNameForFieldIndex(ccColumn) << " ";
```

Comment:
I get: 
VALUES ("mixxx.db.model.playlist.ColumnsSorting"," 1  -1  -1 ","0","0");
this seams to be related to the first columns where ccColumn < 0;